### ISSUE TITLE
ObjectDisposedException thrown when adding or deleting items to/from ItemsSource after reopening a view

### ISSUE THREAD

**Issue by jsuvanto** [roubachof/Sharpnado.Presentation.Forms#54]
**Platform (please complete the following information):**
 - OS: Android
 - Device: Android Emulator 8.1
 - Sdk version: Android SDK 27 and 28
 - Xamarin.Forms: 3.6.0.344457
 - Package version: NuGet Sharpnado.Forms.HorizontalListView v1.1.0

**Describe the bug**
Adding an item to or removing an item from `ItemsSource` (type `ObservableCollection<T>`) throws `System.ObjectDisposedException` if the `HorizontalListView` has been rendered at least once before. I tried compiling using both Android SDK 27 (Oreo 8.1) and 28 (Pie 9.0). Both end up throwing the same exception.

I have made a [demo application](http://jsuvanto.iki.fi/SharpnadoDemo.zip) that always crashes if following the steps below. In it, `MainPage` uses `Navigation.PushModalAsync()` to open a new `ListPage`, which contains the `HorizontalListView`. If this `ListPage` is popped from the UI stack (by using device's back button or `Navigation.PopModalAsync()`) and reopened, any modification to the ItemsSource will cause an exception.

Behaviour is the same whether using the same `ListPage` or always creating a new one.

Related to #34.

**To Reproduce**
1. Download and extract demo application 
1. Build and run application in emulator or Android device
2. Click/Tap "Open list" button to open the list page
3. Click/Tap "Add" and "Delete" buttons to see working behaviour
4. Use emulator's or device's "Back" button to return to main page
5. Click/Tap "Open list" button again
6. Click/Tap "Add" or "Delete" button to cause an exception

**Exceptions (if applicable)**
```
UNHANDLED EXCEPTION:
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'Sharpnado.Presentation.Forms.Droid.Renderers.HorizontalList.SlowRecyclerView'.
  at Java.Interop.JniPeerMembers.AssertSelf (Java.Interop.IJavaPeerable self) [0x00029] in <286213b9e14c442ba8d8d94cc9dbec8e>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00000] in <286213b9e14c442ba8d8d94cc9dbec8e>:0 
  at Android.Support.V7.Widget.RecyclerView.InvalidateItemDecorations () [0x0000a] in <7cd821e41c6245e986359cd08947256b>:0 
  at Sharpnado.Presentation.Forms.Droid.Renderers.HorizontalList.AndroidHorizontalListViewRenderer+RecycleViewAdapter.OnCollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00054] in D:\Dev\Sharpnado\src\Xamarin-Forms-Practices\Sharpnado.Presentation.Forms\Sharpnado.Presentation.Forms.Droid\Renderers\HorizontalList\AndroidHorizontalListViewRenderer.RecycleViewAdapter.cs:317 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_NotifyCollectionChangedEventArgs(object,System.Collections.Specialized.NotifyCollectionChangedEventArgs)
  at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00018] in <fb6d78e506844b3b96d5b35aa047fbbd>:0 
  at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedAction action, System.Object item, System.Int32 index) [0x00009] in <fb6d78e506844b3b96d5b35aa047fbbd>:0 
  at System.Collections.ObjectModel.ObservableCollection`1[T].InsertItem (System.Int32 index, T item) [0x0001a] in <fb6d78e506844b3b96d5b35aa047fbbd>:0 
  at System.Collections.ObjectModel.Collection`1[T].Add (T item) [0x00020] in <d4a23bbd2f544c30a48c44dd622ce09f>:0 
  at SharpnadoDemo.ViewModel.AddRandomString () [0x00010] in C:\Users\juhsuv\source\repos\SharpnadoDemo\SharpnadoDemo\SharpnadoDemo\ViewModel.cs:27 
  at SharpnadoDemo.ListPage.AddButton_Clicked (System.Object sender, System.EventArgs e) [0x00001] in C:\Users\juhsuv\source\repos\SharpnadoDemo\SharpnadoDemo\SharpnadoDemo\ListPage.xaml.cs:23 
  at Xamarin.Forms.Button.Xamarin.Forms.Internals.IButtonElement.PropagateUpClicked () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Button.cs:166 
  at Xamarin.Forms.ButtonElement.ElementClicked (Xamarin.Forms.VisualElement visualElement, Xamarin.Forms.Internals.IButtonElement ButtonElementManager) [0x0001f] in D:\a\1\s\Xamarin.Forms.Core\ButtonElement.cs:61 
  at Xamarin.Forms.Button.SendClicked () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Button.cs:152 
  at Xamarin.Forms.Platform.Android.ButtonElementManager.OnClick (Xamarin.Forms.VisualElement element, Xamarin.Forms.IButtonController buttonController, Android.Views.View v) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\ButtonElementManager.cs:25 
  at Xamarin.Forms.Platform.Android.AppCompat.ButtonRenderer.Android.Views.View.IOnClickListener.OnClick (Android.Views.View v) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\ButtonRenderer.cs:185 
  at Android.Views.View+IOnClickListenerInvoker.n_OnClick_Landroid_view_View_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_v) [0x00011] in <b781ed64f1d743e7881ac038e0fbdf85>:0 
  at (wrapper dynamic-method) System.Object.30(intptr,intptr,intptr)
```

**Comment by roubachof** [comment]
issue is, when closing the current page with PopModal, the ElementChanged method is not called. And recyclerview cleanup is not done.
Had to hack the RecyclerView to wipe zombie event handler.

**Comment by roubachof** [comment]
Thanks again for the detailed issue and the repro sample, it was a breaze to reproduce x)

**Comment by roubachof** [comment]
fixed in v1.2.0

**Comment by jocontacter** [comment]
Reproduced in 1.8.0 with XF 4.8.0.1687:
[mono] Unhandled Exception:
[mono] System.ObjectDisposedException: Cannot access a disposed object.
[mono] Object name: 'Sharpnado.HorizontalListView.Droid.Renderers.HorizontalList.SlowRecyclerView'.
[mono]   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
[mono]   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)

the same circumstances (after reopening view)
I have in my VM:

`public ObservableCollection<Workspace> Items { get; } = new ObservableCollection<Workspace>();`
`public void OnAppearing()
  {
      LoadCommand?.Execute().Subscribe();
  }`

into the LoadCommand i do clear() and reload one by one
